src/common/core/cachedmetacore.js

```diff
@@ -0,0 +1,363 @@
+/*globals define*/
+/*jshint node: true, browser: true*/
+
+/**
+ * @author kecso / https://github.com/kecso
+ */
+
+define(['common/util/assert', 'common/core/core', 'common/core/tasync'], function (ASSERT, Core, TASYNC) {
+        'use strict';
+
+        var MetaCacheCore = function (oldcore, options) {
+            ASSERT(typeof options === 'object');
+            ASSERT(typeof options.globConf === 'object');
+            ASSERT(typeof options.logger !== 'undefined');
+            // copy all operations
+            var core = {},
+                META_SET_NAME = 'MetaAspectSet',
+                logger = options.logger.fork('meta');
+            for (var key in oldcore) {
+                core[key] = oldcore[key];
+            }
+            logger.debug('initialized');
+
+            function loadMetaSet(root) {
+                var paths = oldcore.getMemberPaths(root, META_SET_NAME),
+                    i,
+                    metaElements = [];
```

Comment:
Minor thing, why metaElements and not metaNodes?
getAllMetaNodes returns with metaElements..